app/src/lib/stores/helpers/background-fetcher.ts

```diff
@@ -90,7 +92,12 @@ export class BackgroundFetcher {
       return
     }
 
-    const shouldFetch = this.shouldPerformFetch(this.repository)
+    const shouldFetch = await this.shouldPerformFetch(this.repository)
+
+    if (this.stopped) {
```

Comment:
Had to add the because of the newly introduced async call above. We could have been stopped in between calling that method and it returning.

Reply:
Nice catch!